src/hera/workflows/_runner/script_annotations_util.py

```diff
@@ -95,40 +95,54 @@ def get_annotated_output_param(param_annotation: Parameter) -> Path:
     return path
 
 
-def get_annotated_artifact_value(artifact_annotation: Artifact) -> Union[Path, Any]:
+def get_annotated_output_artifact(artifact_annotation: Artifact) -> Path:
+    if artifact_annotation.path:
+        path = Path(artifact_annotation.path)
+    else:
+        path = _get_outputs_path(artifact_annotation)
+    # Automatically create the parent directory (if required)
+    path.parent.mkdir(parents=True, exist_ok=True)
+    return path
+
+
+def get_annotated_artifact_value(artifact_annotation: Artifact, func_param_annotation: type) -> Union[Path, Any]:
     """Get the value of the given Artifact annotation.
 
     If the artifact is an output, return the path it will write to.
-    If the artifact is an input, return the loaded value (json, path or string) using its ArtifactLoader.
+    If the artifact is an input, return the  loaded value (json obj, path or string)
+    using its ArtifactLoader, unless ArtifactLoader is a Callable, in which case it
+    is used to load the value.
```

Comment:
I don't understand from this text what is different between the if and the else -- they both seem to say it will be loaded using the ArtifactLoader